pinot-core/src/main/java/org/apache/pinot/core/util/SegmentProcessorAvroUtils.java

```diff
@@ -83,45 +88,90 @@ public static Schema convertPinotSchemaToAvroSchema(org.apache.pinot.spi.data.Sc
     for (FieldSpec fieldSpec : orderedFieldSpecs) {
       String name = fieldSpec.getName();
       DataType storedType = fieldSpec.getDataType().getStoredType();
+
+      SchemaBuilder.BaseFieldTypeBuilder<Schema> type;
+      if (fieldSpec.isNullable()) {
+        type = fieldAssembler.name(name).type().nullable();
+      } else {
+        type = fieldAssembler.name(name).type();
+      }
+
+      String logicalType = "pinot." + fieldSpec.getDataType().toString().toLowerCase(Locale.US);
       if (fieldSpec.isSingleValueField()) {
         switch (storedType) {
           case INT:
-            fieldAssembler = fieldAssembler.name(name).type().intType().noDefault();
+            fieldAssembler = type.intBuilder()
+                .prop("logicalType", logicalType)
+                .endInt()
+                .noDefault();
             break;
           case LONG:
-            fieldAssembler = fieldAssembler.name(name).type().longType().noDefault();
+            fieldAssembler = type.longBuilder()
+                .prop("logicalType", logicalType)
+                .endLong()
+                .noDefault();
             break;
           case FLOAT:
-            fieldAssembler = fieldAssembler.name(name).type().floatType().noDefault();
+            fieldAssembler = type.floatBuilder()
+                .prop("logicalType", logicalType)
+                .endFloat()
+                .noDefault();
             break;
           case DOUBLE:
-            fieldAssembler = fieldAssembler.name(name).type().doubleType().noDefault();
+            fieldAssembler = type.doubleBuilder()
+                .prop("logicalType", logicalType)
+                .endDouble()
+                .noDefault();
             break;
           case STRING:
-            fieldAssembler = fieldAssembler.name(name).type().stringType().noDefault();
+          case BIG_DECIMAL:
```

Comment:
Seems `Decimal` is a valid Avro logical type per this [spec](https://avro.apache.org/docs/1.10.2/spec.pdf)